packages/design-system/src/helpers/__tests__/handleResponsiveValues.test.ts

```diff
@@ -0,0 +1,183 @@
+      "padding-block-start: 1rem;
+      padding-inline-end: 1rem;
+      padding-block-end: 1rem;
+      padding-inline-start: 1rem;
+      margin-block-start: 4px;
+      margin-inline-end: 4px;
+      margin-block-end: 4px;
+      margin-inline-start: 4px;"
+    `);
+  });
+
+  it('should handle simple space values from theme', () => {
+    const values = { padding: 2, margin: 4 };
+    expect(handleResponsiveValues(values, lightTheme)).toMatchInlineSnapshot(`
+      "padding-block-start: 8px;
+      padding-inline-end: 8px;
+      padding-block-end: 8px;
+      padding-inline-start: 8px;
+      margin-block-start: 16px;
+      margin-inline-end: 16px;
+      margin-block-end: 16px;
+      margin-inline-start: 16px;"
+    `);
+  });
+
+  it('should handle responsive values with different breakpoints', () => {
+    const values = {
+      padding: {
+        initial: 1,
+        small: 2,
+        medium: 3,
+        large: 4,
+      },
+      margin: {
+        initial: 2,
+        small: 4,
+        medium: 6,
+        large: 8,
+      },
+    };
+    expect(handleResponsiveValues(values, lightTheme)).toMatchInlineSnapshot(`
+      "padding-block-start: 4px;
+      padding-inline-end: 4px;
+      padding-block-end: 4px;
+      padding-inline-start: 4px;
+      margin-block-start: 8px;
+      margin-inline-end: 8px;
+      margin-block-end: 8px;
+      margin-inline-start: 8px;
+      @media(min-width: 520px){ padding-block-start: 8px;
+      padding-inline-end: 8px;
+      padding-block-end: 8px;
+      padding-inline-start: 8px;
+      margin-block-start: 16px;
+      margin-inline-end: 16px;
+      margin-block-end: 16px;
+      margin-inline-start: 16px; }
+      @media(min-width: 768px){ padding-block-start: 12px;
+      padding-inline-end: 12px;
+      padding-block-end: 12px;
+      padding-inline-start: 12px;
+      margin-block-start: 24px;
+      margin-inline-end: 24px;
+      margin-block-end: 24px;
+      margin-inline-start: 24px; }
+      @media(min-width: 1280px){ padding-block-start: 16px;
+      padding-inline-end: 16px;
+      padding-block-end: 16px;
+      padding-inline-start: 16px;
+      margin-block-start: 40px;
+      margin-inline-end: 40px;
+      margin-block-end: 40px;
+      margin-inline-start: 40px; }"
+    `);
+  });
+
+  it('should handle responsive values with mixed prop type', () => {
+    const values = {
+      paddingRight: {
+        initial: 2,
+        small: 4,
+        medium: 6,
+        large: '8px',
+      },
+      padding: {
+        initial: 1,
+        small: 2,
+        medium: '1rem',
+        large: 4,
+      },
+      margin: 2,
+      marginTop: {
+        initial: 1,
+        small: '2rem',
+      },
+    };
+    expect(handleResponsiveValues(values, lightTheme)).toMatchInlineSnapshot(`
+      "padding-inline-end: 4px;
+      padding-block-start: 4px;
+      padding-block-end: 4px;
+      padding-inline-start: 4px;
```

Comment:
i would expect `padding-inline-end` to read `8px` here because we've used `paddingRight`? 🤔

Reply:
Correct, CSS specificity is not taken care here so it wont work, we have to change the order for tests to work